src/main/java/taxi/dao/DriverDaoImpl.java

```diff
@@ -105,10 +110,24 @@ private Driver parseDriverFromResultSet(ResultSet resultSet) throws SQLException
         Long id = resultSet.getObject("id", Long.class);
         String name = resultSet.getString("name");
         String licenseNumber = resultSet.getString("license_number");
+        String login = resultSet.getString("login");
+        String password = resultSet.getString("password");
         Driver driver = new Driver();
         driver.setId(id);
         driver.setName(name);
         driver.setLicenseNumber(licenseNumber);
+        driver.setLogin(login);
+        driver.setPassword(password);
         return driver;
     }
+
+    @Override
+    public Optional<Driver> findByLogin(String login) {
+        for (Driver driver : getAll()) {
+            if (driver.getLogin().equals(login)) {
+                return Optional.of(driver);
+            }
+        }
+        return Optional.empty();
+    }
 }
```

Comment:
maybe it's better to create a method and implement it using the select statements? 
You will avoid redundant for - loop